applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -571,6 +661,41 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateRightH
     noalias(rRightHandSideVector) = - data.vol * density * prod(data.DN_DX, velocity);
 }
 
+template <int TDim, int TNumNodes>
+void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateRightHandSideExtendedElement(
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rRightHandSideVector.size() != TNumNodes + 1)
+    {
+        rRightHandSideVector.resize(TNumNodes + 1, false);
+    }
+    rRightHandSideVector.clear();
+
+    ElementalData<TNumNodes, TDim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    const double density = ComputeDensity(rCurrentProcessInfo);
+
+    // Computing local velocity
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+    array_1d<double, TDim> velocity = PotentialFlowUtilities::ComputeVelocity<TDim,TNumNodes>(*this);
+    for (unsigned int i = 0; i < TDim; i++)
+    {
+        velocity[i] += free_stream_velocity[i];
+    }
```

Comment:
By the way, I noticed that we repeat these lines 5 times in the element. It might make sense to implement a function ComputeTotalVelocity in another PR in the utilities that allows us to substitute them by one line. Something like:

```cpp
template <int Dim, int NumNodes>
array_1d<double, Dim> ComputeTotalVelocity(const Element& rElement, const ProcessInfo& rCurrentProcessInfo)
{
    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];

    array_1d<double, Dim> velocity = ComputeVelocity<Dim,NumNodes>(rElement);

    for (unsigned int i = 0; i < Dim; i++){
        velocity[i] += free_stream_velocity[i];
    }

    return velocity;
}
```